src/main/java/org/folio/service/file/SplitFileProcessingService.java

```diff
@@ -0,0 +1,133 @@
+    JobExecution parentJobExecution,
+    ChangeManagerClient client,
+    int parentJobSize,
+    String tenant,
+    List<String> keys
+  ) {
+    List<Future<JobExecutionDto>> futures = new ArrayList<>();
+
+    int partNumber = 1;
+    for (String key : keys) {
+      InitJobExecutionsRqDto initJobExecutionsRqDto = new InitJobExecutionsRqDto()
+        .withFiles(Arrays.asList(new File().withName(key)))
+        .withParentJobId(parentJobExecution.getId())
+        .withJobPartNumber(partNumber)
+        .withTotalJobParts(keys.size())
+        .withSourceType(InitJobExecutionsRqDto.SourceType.COMPOSITE)
+        .withUserId(
+          Objects.nonNull(parentUploadDefinition.getMetadata())
+            ? parentUploadDefinition.getMetadata().getCreatedByUserId()
+            : null
+        );
+
+      // outer scope variable could change before lambda execution, so we make it final here
+      final int thisPartNumber = partNumber;
+
+      Promise<JobExecutionDto> promise = Promise.promise();
+      futures.add(promise.future());
+
+      client.postChangeManagerJobExecutions(
+        initJobExecutionsRqDto,
+        response -> {
+          try {
+            if (
+              response.result().statusCode() != HttpStatus.HTTP_CREATED.toInt()
+            ) {
+              LOGGER.warn(
+                "registerSplitFiles:: Error creating new child JobExecution for key {}. Status message: {}",
+                key,
+                response.result().statusMessage()
+              );
+              throw new HttpException(
+                response.result().statusCode(),
+                "Error creating new JobExecution"
+              );
+            } else {
+              BufferMapper
+                .mapBufferContentToEntity(
+                  response.result().bodyAsBuffer(),
+                  JobExecutionDtoCollection.class
+                )
+                .map(collection -> collection.getJobExecutions().get(0))
+                .compose(execution ->
+                  queueItemDao
+                    .addQueueItem(
+                      new DataImportQueueItem()
+                        .withJobExecutionId(execution.getId())
+                        .withUploadDefinitionId(parentUploadDefinition.getId())
+                        .withTenant(tenant)
+                        .withOriginalSize(parentJobSize)
+                        .withFilePath(key)
+                        .withTimestamp(Instant.now().toString())
+                        .withPartNumber(thisPartNumber)
+                        .withProcessing(false)
+                    )
+                    .onSuccess(v -> promise.complete(execution))
+                )
+                .onFailure(promise::fail);
+            }
+          } catch (Exception e) {
+            promise.fail(e);
+          }
+        }
+      );
+      partNumber++;
+    }
+
+    return CompositeFuture.join(
+      futures.stream().map(Future.class::cast).collect(Collectors.toList())
```

Comment:
just for my information, why was the cast necessary?

Reply:
The version of vert.x that we're on has some funky behavior with `CompositeFuture` where `CompositeFuture` only takes untyped `Future`s (or in other words, every future used in it must be a plain `Future`, not `Future<Anything>`, and also when we retrieve the values they'll need casting too (otherwise they're stuck as plain `Object`).

Reply:
Noah, there is a dedicated util class GenericCompositeFuture to create CompositeFuture by passing a list of parameterized `Future<Anything>` if you prefer we can use it here.